### PR TITLE
Adding Dapr Spring Boot Starter.

### PR DESCRIPTION
**Pull request by darren-wang**
# Description

Adding a spring boot starter for dapr to better integrate with Java spring boot applicaitons.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/java-sdk/issues/441

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: dapr-spring-boot-starter/src/main/java/io/dapr/springboot/DaprAutoConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Dapr's Spring Boot AutoConfiguration.
+ */
+@Configuration
+@ConditionalOnWebApplication
+@ComponentScan("io.dapr.springboot")
```
Comment (artursouza) [inline dapr-spring-boot-starter/src/main/java/io/dapr/springboot/DaprAutoConfiguration.java:17]: Is this the only package to be scanned?

File: pom.xml [hunk 2]
```diff
@@ -271,6 +271,7 @@
     <module>sdk-actors</module>
     <module>sdk-springboot</module>
     <module>examples</module>
+    <module>dapr-spring-boot-starter</module>
```
Comment (artursouza) [inline pom.xml:274]: sdk-spring-boot-starter

**Review by artursouza — CHANGES_REQUESTED** [review]
Please, modify at least one example showcasing how this can be used.

After that we will go into more details on how to get this PR merged, specially if we need to publish a new maven package.

**Comment by artursouza** [comment]
@darren-wang  Thanks for your contribution. Please, see comments above. This has potential for the springboot community. It would be great to see an example on how. Maybe change one of our examples in this repo demonstrating that.

File: dapr-spring-boot-starter/src/main/java/io/dapr/springboot/DaprAutoConfiguration.java (hunk 1 above)
Comment (darren-wang) [reply dapr-spring-boot-starter/src/main/java/io/dapr/springboot/DaprAutoConfiguration.java:17]: Yes, it's the package where the Spring beans `DaprBeanPostProcessor` and `DaprController` locate, they are the beans provided by dapr to be scanned by spring boot.

**Comment by darren-wang** [comment]
Also, the existing artifact `dapr-sdk-springboot` may be better named with `dapr-sdk-spring` or so, it does not provide spring boot features. By saying spring boot we usually mean auto-configuration and so on. Or, we can simply change `dapr-sdk-springboot` to a boot starter without adding a new starter module, however, it's not the best way to work with traditional non-boot applicaitons, since they may not need boot features.

**Comment by darren-wang** [comment]
Should be replaced by https://github.com/dapr/java-sdk/pull/459

**Pull request closed without merging**